File: services/rust/axum/src/routes/middleware/log.rs
use axum::{extract::Request, middleware::Next, response::Response};
use tracing::{debug, instrument};

#[instrument(skip_all)]
pub async fn log_middleware(req: Request, next: Next) -> Response {
    debug!(method = %req.method(), path = %req.uri(), "Incoming request");

    next.run(req).await
}

File: services/typescript/express/src/routes/middleware/log.ts
import type { RequestHandler, Request, Response, NextFunction } from "express";
import { getLogger } from "telemetry";

export const logMiddleware: RequestHandler = (
  req: Request,
  _res: Response,
  next: NextFunction
) => {
  let method = req.method;
  let path = req.path;

  const logger = getLogger();

  logger.debug({ method, path }, "Incoming request");

  return next();
};
